### PR TITLE
Clarify need to save ConfigFile to apply changes to file

### PR DESCRIPTION
**Pull request by georgespatton**
To make clear documentation based on this: https://github.com/godotengine/godot/issues/52645

### CODE REVIEW

File: doc/classes/ConfigFile.xml [hunk 1]
```diff
@@ -100,15 +100,15 @@
 			<return type="void" />
 			<argument index="0" name="section" type="String" />
 			<description>
-				Deletes the specified section along with all the key-value pairs inside. Raises an error if the section does not exist.
+				Deletes the specified section along with all the key-value pairs inside. Raises an error if the section does not exist.  After calling this method, make sure to call the save() method to write the change to the file.
```
Comment (Calinou) [inline doc/classes/ConfigFile.xml:103]:
```suggestion
				Deletes the specified section along with all the key-value pairs inside. Raises an error if the section does not exist. After calling [method erase_section], make sure to call [method save] method to write the change to the file.
```

File: doc/classes/ConfigFile.xml [hunk 2]
```diff
@@ -100,15 +100,15 @@
...
 			</description>
 		</method>
 		<method name="erase_section_key">
 			<return type="void" />
 			<argument index="0" name="section" type="String" />
 			<argument index="1" name="key" type="String" />
 			<description>
-				Deletes the specified key in a section. Raises an error if either the section or the key do not exist.
+				Deletes the specified key in a section. Raises an error if either the section or the key do not exist.  After calling this method, make sure to call the save() method to write the change to the file.
```
Comment (Calinou) [inline doc/classes/ConfigFile.xml:111]:
```suggestion
				Deletes the specified key in a section. Raises an error if either the section or the key do not exist. After calling [method erase_section_key], make sure to call [method save] method to write the change to the file.
```

**Review by Calinou — DISMISSED** [review]

**Comment by mhilbrunner** [comment]
It should be either `to call [method save]`, not `to call [method save] method` in both, or "to call _the_ [method save] method".

Also the commits need to be squashed into one.

Looks good otherwise, thanks :)

**Comment by akien-mga** [comment]
Isn't this something which should be explained in the main description of the class? AFAIK, any operation that mutates the ConfigFile only changes what is loaded in memory - if you want to apply any change to a file (not just section removals), you have to `save()`.

**Comment by georgespatton** [comment]
> Isn't this something which should be explained in the main description of the class? AFAIK, any operation that mutates the ConfigFile only changes what is loaded in memory - if you want to apply any change to a file (not just section removals), you have to `save()`.

Thanks for the feedback.  I have updated the PR with the explanation in the main description of the class.

**Comment by akien-mga** [comment]
Looks great! Could you squash the commits together? See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase) for instructions.

**Comment by akien-mga** [comment]
Also note that the Git identity you used to author that last commit (probably with local Git) is not configured properly, as your email is `root@localhost`, which is not valid (see header of https://github.com/godotengine/godot/commit/4f670e0d6d25444afe5931164595a17baa47f947.patch).

You might want to change it to a valid email linked to your GitHub account to get proper attribution for your commits (see https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-email-preferences/setting-your-commit-email-address, which also has an option for a "no-reply" email if you prefer).

**Comment by georgespatton** [comment]
> Also note that the Git identity you used to author that last commit (probably with local Git) is not configured properly, as your email is `root@localhost`, which is not valid (see header of https://github.com/godotengine/godot/commit/4f670e0d6d25444afe5931164595a17baa47f947.patch).
> 
> You might want to change it to a valid email linked to your GitHub account to get proper attribution for your commits (see https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-email-preferences/setting-your-commit-email-address, which also has an option for a "no-reply" email if you prefer).

I think I did this right.  Let me know if there are any additional adjustments that you would like to see.

**Comment by akien-mga** [comment]
It's not right yet, as you can see in the PR there are now 6 commits:

Screenshot_20210914_144815

Most likely, after squashing the commits you did not use `git push --force` to update the remote branch, but you pulled/merged the remote branch first to resolve the history conflicts.

I'll push an update to resolve the situation as it's a bit tricky now if you're not familiar with Git rebase.

**Review by akien-mga — APPROVED** [review]
Looks good to me.

I squashed the commits and edited the commit log (and wording) to make changes more explicit.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by georgespatton** [comment]
> Thanks! And congrats for your first merged Godot contribution

Thanks for the assist.  I'll not be such a noob next commit.

**Comment by mhilbrunner** [comment]
No worries, git is a complicated beast and we all needed to learn it :)

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.

**Comment by akien-mga** [comment]
Cherry-picked for 3.3.4.

**Pull request merged**